packages/api-reference/src/components/ApiReference.vue

```diff
@@ -229,91 +222,36 @@ watch(
   { immediate: true },
 )
 
-onMounted(() => {
-  document.querySelector('#tippy')?.scrollTo({
-    top: 0,
-    left: 0,
-  })
+const swaggerEditorRef = ref<typeof SwaggerEditor | undefined>()
+const { toggleDarkMode, isDark } = useDarkModeState()
 
+onMounted(() => {
   fetchSpecFromUrl()
 })
-
-const showRenderedContent = computed(
-  () => isLargeScreen.value || !currentConfiguration.value?.isEditable,
-)
-
-const showSwaggerEditor = computed(() => {
-  return (
-    !currentConfiguration.value.spec?.preparsedContent &&
-    currentConfiguration.value?.isEditable
-  )
-})
-
-function handleAIWriter(
-  value: string[],
-  swaggerData: string,
-  swaggerType: 'json' | 'yaml',
-) {
-  emits('startAIWriter', value, swaggerData, swaggerType)
-}
 </script>
 <template>
   <ThemeStyles :id="currentConfiguration?.theme" />
   <FlowToastContainer />
-  <div
-    ref="documentEl"
-    class="scalar-api-reference references-layout"
-    :class="[
-      {
-        'references-footer-below': currentConfiguration?.footerBelowSidebar,
-        'references-editable': showSwaggerEditor,
-      },
-    ]"
-    :style="{ '--full-height': `${elementHeight}px` }">
-    <slot name="search-modal">
-      <SearchModal
-        :spec="parsedSpec"
-        variant="search" />
-    </slot>
-    <!-- Desktop header -->
-    <div
-      v-if="!isMobile"
-      class="references-header">
-      <slot name="header"></slot>
-    </div>
-    <!-- Navigation (sidebar) wrapper -->
-    <aside class="references-navigation t-doc__sidebar">
-      <!-- Mobile header content -->
-      <slot
-        v-if="isMobile"
-        :label="state.activeBreadcrumb"
-        name="mobile-header">
-        <!-- Fallback mobile header -->
-        <MobileHeader>
-          {{ state.activeBreadcrumb }}
-        </MobileHeader>
-      </slot>
-      <!-- Navigation tree / Table of Contents -->
-      <!-- Sorry for the terrible v-if - this is so we only manage the menu state if theres no external mobile header being injected to manage it otherwise -->
-      <div
-        v-if="
-          isMobile && !$slots['mobile-header']
-            ? currentConfiguration.showSidebar && showMobileDrawer
-            : currentConfiguration.showSidebar
-        "
-        class="references-navigation-list">
-        <slot
-          v-if="isMobile"
-          name="header" />
-        <Sidebar
-          :searchHotKey="currentConfiguration.searchHotKey"
-          :spec="parsedSpec" />
-      </div>
-    </aside>
-    <!-- Swagger file editing -->
-    <div
-      v-show="showSwaggerEditor"
-      class="references-editor">
+  <ApiReferenceLayout
+    :currentConfiguration="currentConfiguration"
+    :isDarkMode="isDark"
+    :parsedSpec="parsedSpec"
+    :rawSpec="rawSpecRef"
+    :swaggerEditorRef="swaggerEditorRef"
+    @changeTheme="$emit('changeTheme', $event)"
+    @toggleDarkMode="toggleDarkMode">
+    <template #header>
+      <slot name="header" />
```

Comment:
I don't think we need a slot here (@hanspagel what do you think?) but whether we have a slot or not the content should be provided here (either as slot fallback or as direct slot content) same as it was before.

Reply:
I believe this was just copied over from how it was done before 
![image](https://github.com/scalar/scalar/assets/2039539/91ea9437-eedb-4c2c-a8ea-f5bf24981fb4)